### PR TITLE
Tree filtering

### PR DESCRIPTION
**Pull request by albert-smit**
adds filtering to the treeview(issue #53)
also adds a reset button for the filter, and a toString method to defaultSequence

### CODE REVIEW

File: src/main/java/nl/tudelft/lifetiles/tree/controller/TreeController.java [hunk 1]
```diff
@@ -52,6 +74,29 @@ public final void initialize(final URL location,
...
+        });
+
+        listen(Message.FILTERED, (controller, args) -> {
+            // check the message is correct
+                assert args.length == 1;
+                if (!(args[0] instanceof Set<?>)) {
+                    throw new IllegalArgumentException(
+                            "Argument not of type Set<Sequence>");
+                }
+                if (!(controller instanceof TreeController)) {
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/tree/controller/TreeController.java:91]: Indentation

File: src/main/java/nl/tudelft/lifetiles/tree/controller/TreeController.java [hunk 2]
```diff
@@ -52,6 +74,29 @@ public final void initialize(final URL location,
...
+                assert (args[1] instanceof Map<?, ?>);
+                sequences = (Map<String, Sequence>) args[1];
+                repaint();
+            }
+        });
+
+        listen(Message.FILTERED, (controller, args) -> {
+            // check the message is correct
+                assert args.length == 1;
+                if (!(args[0] instanceof Set<?>)) {
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/tree/controller/TreeController.java]: `assert`might be better

File: src/main/java/nl/tudelft/lifetiles/tree/controller/TreeController.java [hunk 3]
```diff
@@ -82,9 +130,108 @@ private void loadTree(final File file) throws FileNotFoundException {
...
+        repaint();
+    }
+
+    /**
+     * Informs other controllers the filter changed.
+     *
+     * @param visible
+     *            the set that needs to be visible
+     */
+    public void shoutVisible(final Set<Sequence> visible) {
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/tree/controller/TreeController.java]: Redundant method

File: src/main/java/nl/tudelft/lifetiles/tree/view/SunburstCenter.java [hunk 4]
```diff
@@ -24,14 +26,29 @@ public SunburstCenter() {
...
+    /**
+     * Creates a {@link Color} for this node. the color will be blue by default,
+     * and the color associated with the sequence when the node has a sequence.
+     *
+     * @return a Color object that specifies what color this node will be.
+     */
+    private Color createColor() {
+        Sequence sequence = getValue().getSequence();
+        if (sequence == null) {
+            return Color.BLUE;
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/tree/view/SunburstCenter.java]: Default color could use a constant

**Comment by joswinter** [comment]
Looks good, if the above points are fixed this can be merged.

**Comment by rutgervandenberg** [comment]
Looks fine.

**Comment by alangerak** [comment]
Looks good

**Pull request merged**